public/app/features/explore/state/actions.ts

```diff
@@ -466,8 +466,9 @@ export function runQueries(exploreId: ExploreId): ThunkResult<void> {
 
     const queryOptions = {
       minInterval,
-      // This is used for logs streaming for buffer size.
-      maxDataPoints: mode === ExploreMode.Logs ? 1000 : containerWidth,
+      // This is used for logs streaming for buffer size, with undefined it falls back to datasource config if it
+      // supports that.
+      maxDataPoints: mode === ExploreMode.Logs ? undefined : containerWidth,
```

Comment:
Is there some way to extract this to be testable?

Reply:
@hugohaggmark not exactly sure what you mean. The Loki datasource actually have test to check whether the `maxDataPoints` option works correctly (and it also tests whether `undefined` works although that does not conform to the typings).

Or do you mean testing whether we get the correct value for the specified `ExploreMode`?

Reply:
No biggie, I was thinking something in the line of `cont getMaxDataPoints = (mode) => ` and then have some simple tests for that.